llvm/include/llvm/ADT/DenseMapInfo.h

```diff
@@ -137,7 +137,10 @@ template<> struct DenseMapInfo<unsigned long> {
   static inline unsigned long getTombstoneKey() { return ~0UL - 1L; }
 
   static unsigned getHashValue(const unsigned long& Val) {
-    return (unsigned)(Val * 37UL);
+    if constexpr (sizeof(Val) == 4)
```

Comment:
Now we have `densemap::detail::mix` with very low latency, we can use it unconditionally for now.

Reply:
I think it's best to keep the conditional so you get the same hashing behavior if you use a different spelling for what is essentially the same type.

Reply:
Yeah, I'd like to keep the condition too.